### PR TITLE
chore: migrate upload plugin admin to typescript

### PR DESCRIPTION
**Pull request by simotae14**
### What does it do?

- Migrate upload plugin admin to v5

### Files to migrate?

- [ ] admin
  - [ ] src
    - [ ] components
      - [x] AssetCard
        - [x] tests
          - [x] AssetCardBase.test.jsx
          - [x] DocAssetCard.test.jsx
          - [x] ImageAssetCard.test.jsx 
        - [x] AssetCard.jsx
        - [x] AssetCardBase.jsx
        - [x] AudioAssetCard.jsx
        - [x] AudioPreview.jsx
        - [x] DocAssetCard.jsx
        - [x] ImageAssetCard.jsx
        - [x] UploadingAssetCard.jsx
        - [x] VideoAssetCard.jsx
        - [x] VideoPreview.jsx 
      - [ ] AssetDialog
        - [x] BrowseStep
          - [x] PaginationFooter
            - [x] components.jsx
            - [x] index.jsx
            - [x] Pagination.jsx
            - [x] PaginationContext.jsx  
        - [x] SearchAsset
          - [x] tests
            - [x] index.test.jsx
          - [x] index.jsx
        - [ ] tests
          - [ ] __snapshots__
          - [ ] index.test.jsx
        - [x] utils
          - [x] tests
            - [x] isSelectable.test.js
          - [x] isSelectable.js     
        - [x] Filters.jsx
        - [ ] index.jsx
        - [x] PageSize.jsx 
        - [x] SelectedStep
          - [x] index.jsx
        - [ ] tests
          - [ ] AssetDialog.test.jsx
        - [x] DialogFooter.jsx
        - [ ] index.jsx  
      - [x] AssetGridList
        - [x] tests
          - [x] __snapshots__
          - [x] AssetGridList.test.jsx 
        - [x] Draggable.jsx
        - [x] index.jsx 
      - [ ] Breadcrumbs
        - [ ] tests
          - [ ] __snapshots__
          - [ ] index.test.jsx 
        - [ ] Breadcrumbs.jsx
        - [ ] CrumbSimpleMenuAsync.jsx
        - [ ] index.jsx
      - [ ] BulkMoveDialog
        - [ ] tests
          - [ ] __snapshots__
          - [ ] BulkMoveDialog.test.jsx 
        - [ ] BulkMoveDialog.jsx
        - [ ] index.jsx 
      - [x] ContextInfo
        - [x] tests
          - [x] __snapshots__
          - [x] ContextInfo.test.jsx 
        - [x] ContextInfo.jsx
        - [x] index.jsx 
      - [x] CopyLinkButton
        - [x] index.jsx 
      - [ ] EditAssetDialog
        - [x] PreviewBox
          - [x] AssetPreview.jsx
          - [x] components.jsx
          - [x] CroppingActions.jsx
          - [x] index.jsx
        - [ ] tests
          - [x] __snapshots__
          - [ ] EditAssetDialog.test.jsx
          - [ ] index.test.jsx
          - [x] RemoveAssetDialog.test.jsx
        - [x] DialogHeader.jsx
        - [ ] index.jsx
        - [x] RemoveAssetDialog.jsx
        - [x] ReplaceMediaButton.jsx     
      - [ ] EditFolderDialog
        - [x] ModalHeader
          - [x] index.jsx
          - [x] ModalHeader.jsx
        - [ ] tests
          - [ ]  __snapshots__
          - [ ] EditFolderDialog.test.jsx
        - [ ] EditFolderDialog.jsx
        - [ ] index.jsx
        - [x] RemoveFolderDialog.jsx
      - [x] EmptyAssets
        - [x] EmptyAssetGrid.jsx
        - [x] index.jsx
      - [x] FilterList
        - [x] tests
          - [x]  __snapshots__
          - [x] index.test.js
        - [x] FilterTag.jsx
        - [x] index.jsx  
      - [x] FilterPopover
        - [x] utils
          - [x] getFilterList.js
        - [x] FilterValueInput.jsx
        - [x] index.jsx 
      - [x] FolderCard
        - [x] contexts
        - [x] FolderCard
          - [x] FolderCard.jsx
          - [x] index.jsx 
        - [x] FolderCardBody
          - [x] FolderCardBody.jsx
          - [x] index.jsx
        - [x] FolderCardBodyAction
          - [x] index.js 
        - [x] FolderCardCheckbox
          - [x] FolderCardCheckbox.jsx
          - [x] index.jsx
        - [x] hooks
          - [x] tests
            - [x] useId.test.js
          - [x] useId.js
        - [x] tests
          - [x] __snapshots__
          - [x] FolderCard.test.jsx 
        - [x] index.jsx
      - [x] FolderGridList
        - [x] tests
          - [x] __snapshots__
          - [x] FolderGridList.test.jsx
        - [x] FolderGridList.jsx
        - [x] index.jsx 
      - [ ] MediaLibraryDialog
        - [ ] index.jsx 
      - [ ] MediaLibraryInput
        - [ ] Carousel
          - [ ] tests
            - [ ] CarouselAssets.test.jsx
          - [x] CarouselAsset.jsx
          - [x] CarouselAssetActions.jsx
          - [ ] CarouselAssets.jsx
          - [x] EmptyStateAsset.jsx 
        - [ ] tests
          - [ ] MediaLibraryInput.test.jsx 
        - [ ] index.jsx
      - [ ] SelectTree
        - [ ] tests
          - [ ] __snapshots__
          - [ ] SelectTree.test.jsx
        - [x] utils
          - [x] tests
            - [x] __snapshots__
            - [x] flattenTree.test.js
            - [x] getOpenValues.test.js
            - [x] getValuesToClose.test.js
          - [x] flattenTree.js
          - [x] getOpenValues.js
          - [x] getValuesToClose.js
        - [ ] index.jsx
        - [x] Option.jsx
        - [ ] SelectTree.jsx
        - [ ] SelectTree.stories.mdx 
      - [x] SortPicker
        - [x] index.jsx
      - [x] TableList
        - [x] tests
          - [x] __snapshots__
          - [x] CellContent.test.jsx
          - [x] TableList.test.jsx
          - [x] TableRows.test.jsx
        - [x] CellContent.jsx
        - [x] index.jsx
        - [x] PreviewCell.jsx
        - [x] TableRows.jsx
      - [ ] UploadAssetDialog
        - [x] AddAssetStep
          - [x] tests
            - [x] __snapshots__
            - [x] FromComputerForm.test.jsx
          - [x] AddAssetStep.jsx
          - [x] FromComputerForm.jsx
          - [x] FromUrlForm.jsx 
        - [x] PendingAssetStep
          - [x] tests
            - [x] __snapshots__
            - [x] PendingAssetStep.test.jsx
          - [x] PendingAssetStep.jsx 
        - [ ] tests
          - [ ] UploadAssetDialog.test.jsx
        - [ ] UploadAssetDialog.jsx 
      - [x] UploadProgress
        - [x] tests
          - [x] index.test.jsx 
        - [x] index.jsx
    - [x] hooks
      - [x] __mocks
        - [x] useAssets.js
        - [x] useBulkMove.js
        - [x] useBulkRemove.js
        - [x] useConfig.js
        - [x] useEditFolder.js
        - [x] useFolder.js
        - [x] useFolders.js
        - [x] useFolderStructure.js
        - [x] useMediaLibraryPermissions.js
        - [x] useModalQueryParams.js 
      - [x] tests
        - [x] useAssets.test.js
        - [x] useBulkMove.test.js
        - [x] useBulkRemove.test.js
        - [x] useClipboard.test.ts
        - [x] useConfig.test.js
        - [x] useEditFolder.test.js
        - [x] useFolder.test.js
        - [x] useFolders.test.js
        - [x] useFolderStructure.test.js
        - [x] useModalQueryParams.test.js
        - [x] usePersistentState.test.ts
        - [x] useRemoveAsset.test.js
        - [x] useSelectionState.test.ts 
      - [x] utils
        - [x] tests
          - [x] rename-keys.test.js 
        - [x] rename-keys.js
      - [x] useAssets.js
      - [x] useBulkMove.js
      - [x] useBulkRemove.js
      - [x] useClipboard.ts
      - [x] useConfig.js
      - [x] useCropImg.js
      - [x] useEditAsset.js
      - [x] useEditFolder.js
      - [x] useFolder.js
      - [x] useFolders.js
      - [x] useFolderStructure.js
      - [x] useMediaLibraryPermissions.js
      - [x] useModalQueryParams.js
      - [x] usePersistentState.ts
      - [x] useRemoveAsset.js
      - [x] useSelectionState.ts
      - [x] useUpload.js 
    - [ ] pages
      - [ ] App
        - [ ] components
          - [x] tests
            - [x] EmptyOrNoPermissions.test.jsx
          - [ ] BulkActions.jsx
          - [x] BulkDelete.jsx
          - [ ] BulkMoveButton.jsx
          - [x] EmptyOrNoPermissions.jsx
          - [ ] Filters.jsx
          - [ ] Header.jsx
        - [x] ConfigureTheView
          - [x] components
            - [x] tests
              - [x] __snapshots 
            - [x] Settings.jsx
          - [x] state
            - [x] tests
              - [x] init.test.js
              - [x] reducer.test.js
            - [x] actions.js
            - [x] actionTypes.js
            - [x] init.js
            - [x] reducer.js 
          - [x] tests
            - [x] __snapshots
            - [x] ConfigureTheView.test.jsx
          - [x] index.jsx
        - [ ] MediaLibrary
          - [ ] components
            - [ ] tests
              - [ ] __snapshots__
              - [ ] BulkActions.test.jsx
              - [x] BulkDeleteButton.test.jsx
              - [ ] BulkMoveButton.test.jsx
              - [x] EmptyOrNoPermissions.test.jsx
              - [ ] Header.test.jsx
            - [ ] BulkActions.jsx
            - [x] BulkDeleteButton.jsx
            - [ ] BulkMoveButton.jsx
            - [x] EmptyOrNoPermissions.jsx
            - [ ] Filters.jsx
            - [ ] Header.jsx
          - [ ] tests
            - [ ] MediaLibrary.test.jsx
          - [ ] index.jsx
        - [ ] index.jsx 
      - [x] SettingsPage
        - [x] tests
          - [x] index.test.jsx
          - [x] reducer.test.js 
        - [x] index.jsx
        - [x] init.js
        - [x] reducer.js
    - [x] translations
      - [x] tests
        - [x] plural.test.js
    - [x] utils
      - [x] tests
        - [x] appendSearchParamsToUrl.test.js
        - [x] containsAssetFilter.test.js
        - [x] downloadFile.test.js
        - [x] findRecursiveFolderByValue.test.js
        - [x] findRecursiveFolderMetadatas.test.js
        - [x] formatBytes.test.js
        - [x] formatDuration.test.js
        - [x] getAllowedFiles.test.js
        - [x] getAPIInnerErrors.test.js
        - [x] getBreadcrumbDataCM.test.js
        - [x] getBreadcrumbDataML.test.js
        - [x] getFileExtension.test.js
        - [x] getFolderParents.test.js
        - [x] getFolderURL.test.js
        - [x] normalizeAPIError.test.js
        - [x] prefixFileUrlWithBackendUrl.test.js
        - [x] prefixPluginTranslations.test.js
        - [x] toSingularTypes.test.js
        - [x] typeFromMime.test.js
      - [x] appendSearchParamsToUrl.js
      - [x] containsAssetFilter.js
      - [x] createAssetUrl.js
      - [x] displayedFilters.js
      - [x] downloadFilte.js
      - [x] findRecursiveFolderByValue.js
      - [x] findRecursiveFolderMetadatas.js
      - [x] formatBytes.js
      - [x] formatDuration.js
      - [x] getAllowedFiles.js
      - [x] getAPIInnerErrors.js
      - [x] getBreadcrumbDataCM.js
      - [x] getBreadcrumbDataML.js
      - [x] getFileExtension.js
      - [x] getFolderParents.js
      - [x] getFolderURL.js
      - [x] getTrad.js
      - [x] index.js
      - [x] moveElement.js
      - [x] normalizeAPIError.js
      - [x] prefixFileUrlWithBackendUrl.js
      - [x] prefixPluginTranslations.js
      - [x] rawFileToAsset.js
      - [x] toSingularTypes.js
      - [x] typeFromMime.js
      - [x] urlsToAssets.js
      - [x] urlYupSchema.js
    - [ ] constants.js
    - [ ] index.js
    - [x] pluginId.js 
  - [x] tests  
    - [x] handlers.js
    - [x] setup.js
    - [x] utils.jsx 
- [x] documentation 
  - [x] content-api.json
- [x] .eslintignore
- [x] .eslintrc
- [x] .gitattributes
- [x] .gitignore
- [x] .npmignore
- [x] jest.config.front.js
- [x] jest.config.js
- [x] jsconfig.json
- [x] LICENSE
- [ ] package.json
- [ ] packup.config.ts
- [x] README.md
- [x] strapi-server.js 

Define the 
 - [x] Contracts

### CODE REVIEW

**Review by madhurisandbhor — COMMENTED** [review]
Wouldn't it be more convenient(for both parties) to raise separate PR and merge in this main PR with all file details.

File: packages/core/upload/admin/src/utils/toSingularTypes.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export const toSingularTypes = (types: string[]) => {
```
[the remark this answers is not in the document]
Comment (simotae14) [reply packages/core/upload/admin/src/utils/toSingularTypes.ts:1]: if you take a look to the unit tests and also the places where this util is used you can see it takes an array of string and return the same

File: packages/core/upload/admin/src/components/AssetCard/tests/AssetCardBase.test.tsx [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
 import { IntlProvider } from 'react-intl';
 
-import { AssetCardBase } from '../AssetCardBase';
+import { AssetCardBase, AssetCardBaseProps } from '../AssetCardBase';
 
-const render = (props) => ({
+const render = (props: AssetCardBaseProps) => ({
   user: userEvent.setup(),
-  ...renderRTL(<AssetCardBase name="Card" extension="png" {...props} />, {
+  ...renderRTL(<AssetCardBase {...props} />, {
```
[the remark this answers is not in the document]
Comment (simotae14) [reply packages/core/upload/admin/src/components/AssetCard/tests/AssetCardBase.test.tsx:10]: I need this because passing them as props gave me an error

**Comment by simotae14** [comment]
I understand your point but to me was difficult to find a proper way to split the work in different branches, and also the other migrations were done in a single main branch

File: packages/core/upload/shared/contracts/files.ts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+import { errors } from '@strapi/utils';
+
+type SortOrder = 'ASC' | 'DESC';
+
+type SortKey = 'createdAt' | 'name';
+
+/**
+ * GET /upload/files - Get files
+ */
+export interface File {
```
Comment (Marc-Roig) [inline packages/core/upload/shared/contracts/files.ts:14]: A bit different, cause this is what we use to store it on db, but I saw some differences between this contract and the db definition
https://github.com/strapi/strapi/blob/bc9f526e2ba2b0b11ae11df6628346dc58871e98/packages/providers/upload-local/src/index.ts#L10

There are some fields that could be optional here (width, height, caption, alternativeText, ext)

File: packages/core/upload/admin/src/components/AssetCard/tests/AssetCardBase.test.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (simotae14) [reply packages/core/upload/admin/src/components/AssetCard/tests/AssetCardBase.test.tsx:10]: I need it to pass the unit tests, passing them via props I have an issue

**Pull request closed without merging**